pkg/inference/backends/llamacpp/llamacpp_config.go

```diff
@@ -45,7 +57,20 @@ func (c *Config) GetArgs(modelPath, socket string, mode inference.BackendMode) [
 		args = append(args, "--embeddings")
 	}
 
-	return args
+	// Add arguments from model config
+	if modelCfg.ContextSize != nil {
+		args = append(args, "--ctx-size", strconv.FormatUint(*modelCfg.ContextSize, 10))
+	}
+
+	// Add arguments from backend config
+	if config != nil {
+		if config.ContextSize > 0 && !containsArg(args, "--ctx-size") {
+			args = append(args, "--ctx-size", fmt.Sprintf("%d", config.ContextSize))
+		}
+		args = append(args, config.RuntimeFlags...)
+	}
```

Comment:
I would unify the number-to-string conversion; `strconv.Itoa` is probably the most efficient (vs. `strconv.FormatUint` or `fmt.Sprintf`).